applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/high_cycle_fatigue_law_integrator.h

```diff
@@ -0,0 +1,322 @@
+            rMaximumStress = stress_1;
+            rPreviousMaximumStress = rMaximumStress;
+            rCycleCounter = true;
+        } else if (stress_increment_1 <= -0.001 && stress_increment_2 >= 0.0 && !rCycleCounter) {
+            rMinimumStress = stress_1;
+            rPreviousMinimumStress = rMinimumStress;
+            rCycleCounter = true;
+
+        }
+        
+        if (rPreviousMaximumStress != rMaximumStress && rPreviousMinimumStress != rMinimumStress && !rCycleCounter) {
+            rNumberOfCycles++;
+            rPreviousMaximumStress = 0.0;
+            rPreviousMinimumStress = 0.0;
+        }
+    }
+
+    /**
+     * @brief This method checks if the global stress state is tension or compression 
+     * @param StressVector Current predictive stress tensor.
+     * @param Factor Method's output; -1 for a generalized compression state and 1 for a generalized tensile state.
+     */
+    static void CalculateTensionCompressionFactor(const Vector& rStressVector, double& rFactor) 
+    {
+        array_1d<double,3> principal_stresses;
+        ConstitutiveLawUtilities<6>::CalculatePrincipalStresses(principal_stresses, rStressVector);
+
+
+        double abs_component = 0.0, average_component = 0.0, sum_abs = 0.0, sum_average = 0.0;
+        for (unsigned int i = 0; i < principal_stresses.size(); ++i) {
+            abs_component = std::abs(principal_stresses[i]);
+            average_component = 0.5 * (principal_stresses[i] + abs_component);
+            sum_average += average_component;
+            sum_abs += abs_component;
+        }
+        const double pre_indicator = sum_average / sum_abs;
+        if (pre_indicator < 0.5) {
+            rFactor = -1.0;
+        } else {
+            rFactor = 1.0;
+        }
+    }
+
+    /**
+     * @brief This method computes de reversion factor 
+     * @param MaxStress Signed maximum equivalent stress in the current cycle.
+     * @param MinStress Signed minimum equivalent stress in the current cycle. 
+     */
+    static void CalculateReversionFactor(const double MaxStress, const double MinStress, double& rReversionFactor)
+    {
+        rReversionFactor = MinStress / MaxStress;
+    }
+
+    /**
+     * @brief This method computes the reduction factor used in the high cycle fatigue model 
+     * @param MaxStress Signed maximum stress in the current cycle.
+     * @param MinStress Signed minimum stress in the current cycle.
+     * @param ReversionFactor Ratio between the minimum and maximum signed equivalent stresses for the current load cycle.
+     * @param MaterialParameters Material properties.
+     * @param NumbreOfCycles Number of cycles at the current step.
+     * @param FatigueReductionFactor Reduction factor from the previous step to be reevaluated.
+     * @param B0 Internal variable of the fatigue model used for the calculation of the FatigueReductionFactor.
+     */
+    static void CalculateFatigueReductionFactor(const double MaxStress,
```

Comment:
This function name is good, but the function body has other side effects (calculates rReversionFactor and rB0). This is misleading and should probably be split up into smaller functions.

Reply:
I have changed this by taking out the calculation of the ReversionFactor but I have kept B0 inside the function as it is better to compute simultaneously B0 and the fatigue reduction factor. I prefer to change the function name to 'CalculateFatigueReductionFactorAndFatigueReductionParameter( )'. Does it sound fine to you @msandre ?